cmd/kube-scheduler/app/server.go

```diff
@@ -172,34 +173,19 @@ func Run(ctx context.Context, cc schedulerserverconfig.CompletedConfig, outOfTre
 		}
 	}
 
-	if len(cc.ComponentConfig.Profiles) != 1 {
-		// TODO(#85737): Support more than one profile.
-		return errors.New("multiple scheduling profiles are unsupported")
-	}
-	profile := cc.ComponentConfig.Profiles[0]
-	// Prepare event clients.
-	if _, err := cc.Client.Discovery().ServerResourcesForGroupVersion(eventsv1beta1.SchemeGroupVersion.String()); err == nil {
-		cc.Broadcaster = events.NewBroadcaster(&events.EventSinkImpl{Interface: cc.EventClient.Events("")})
-		cc.Recorder = cc.Broadcaster.NewRecorder(scheme.Scheme, profile.SchedulerName)
-	} else {
-		recorder := cc.CoreBroadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: profile.SchedulerName})
-		cc.Recorder = record.NewEventRecorderAdapter(recorder)
-	}
-
+	recorderFactory := getRecorderFactory(&cc)
 	// Create the scheduler.
 	sched, err := scheduler.New(cc.Client,
 		cc.InformerFactory,
 		cc.PodInformer,
-		cc.Recorder,
+		recorderFactory,
 		ctx.Done(),
-		scheduler.WithName(profile.SchedulerName),
+		scheduler.WithProfiles(cc.ComponentConfig.Profiles...),
```

Comment:
My original concern was if the "slice -> unpack via `profiles...` -> pack via `...profiles` -> slice" conversion would cause additional overhead. I wrote a test and have verified it doesn't, so I'm good with the current option - which favors the tests.

Reply:
That's good to know. But even if it did, I don't think customers should have a significant number of profiles. Probably just a handful, and that's already a lot :)